### PR TITLE
[core] add `json` method for http responses

### LINKED ISSUE #16655: HttpResponse should support a .json()
Which is syntactic sugar for :
```python
json.loads(self.text('utf-8-sig'))
```

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #16655

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 1]
```diff
@@ -537,6 +537,16 @@ def text(self, encoding=None):
             encoding = "utf-8-sig"
         return self.body().decode(encoding)
 
+    def json(self):
+        # type: () -> Dict[str, Any]
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:541]: Let's avoid being more constraints than the stdlib lib itself:
https://github.com/python/typeshed/blob/27f38b8c23da0a2d0ff9ce486f84664c8b9a2d58/stdlib/json/__init__.pyi#L46

Any should do it

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 2]
```diff
@@ -537,6 +537,16 @@ def text(self, encoding=None):
...
+        """Return the whole body as a json object.
+
+        :return: The JSON deserialized response body
+        :rtype: dict[str, any]
+        :raises json.decoder.JSONDecodeError or ValueError (in python 2.7):
+        """
+        return json.loads(self.text())
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:548]: Let's pass utf-8-sig as encoding

Comment (iscai-msft) [reply sdk/core/azure-core/azure/core/pipeline/transport/_base.py:548]: ok, makes sense!

**Review by lmazuel — APPROVED** [review]

**Comment by iscai-msft** [comment]
putting on the backburner, will address elsewhere

**Pull request closed without merging**